src/compiler/transformer.ts

```diff
@@ -299,6 +310,31 @@ namespace ts {
             lexicalEnvironmentSuspended = false;
         }
 
+        function hasLeadingComments(node: Node) {
+            node = getParseTreeNode(node);
+            if (!node) {
+                return node && node.emitNode && !!node.emitNode.leadingComments;
+            }
+            const ranges = getLeadingCommentRangesOfNode(node, getSourceFileOfNode(node));
+            return ranges && ranges.length > 0;
+        }
+
+        function chunkVariableDeclarationsBySyntheticComments(declarations: VariableDeclaration[]): VariableDeclaration[][] {
+            return groupBy(declarations, decl => decl.emitNode && decl.emitNode.commentContext && hasLeadingComments(decl.emitNode.commentContext)
+                ? "" + getNodeId(decl.emitNode.commentContext)
+                : "no-comment"
+            );
+        }
+
+        function createSimpleVariableStatement(declarations: VariableDeclaration[]) {
+            const statement = createVariableStatement(/*modifiers*/ undefined, createVariableDeclarationList(declarations));
+            // Bring forward comments on declarations whose statements are omitted due to variable hoisting and place them on the hoisted declaration
+            if (declarations[0] && declarations[0].emitNode && declarations[0].emitNode.commentContext) {

```

Comment:
Comment that since these declarations were grouped by the comment context, `declarations[1].emitNode.commentContext` would be the same thing.